source/DemoView.cpp

```diff
@@ -0,0 +1,179 @@
+//	Destructor
+//---------------------------------------------------
+DemoView::~DemoView()
+{
+	SaveSettings(&Settings);
+}
+
+//---------------------------------------------------
+//	Attached to window - set siblings target to this
+//---------------------------------------------------
+void
+DemoView::AttachedToWindow()
+{
+	Window()->SetPulseRate(Settings.pulse);
+}
+
+//---------------------------------------------------
+//	Draw view
+//---------------------------------------------------
+void
+DemoView::Draw(BRect updateRect)
+{
+	//demo background
+	SetHighColor(255, 255, 255, 255);
+	FillRoundRect(updateRect, 2, 2);
+	SetHighColor(0, 0, 0, 255);
+	StrokeRoundRect(updateRect, 2, 2);
+
+	InitDemo();
+}
+
+//---------------------------------------------------
+//	Draw demo
+//---------------------------------------------------
+void
+DemoView::DrawDemo()
+{
+	BRect rect = Bounds();
+	float x = rect.right / 2;
+	float y = rect.bottom / 2;
+
+	rect.left++;
+	rect.top++;
+	rect.right--;
+	rect.bottom--;
+
+	if (demoCircle < Settings.circles)
+	{
+		float oldPen = PenSize();
+		SetPenSize(Settings.lwidth);
+		BRegion region = BRegion(rect);
+		ConstrainClippingRegion(&region);
+
+		SetHighColor(demoColor);
+		StrokeEllipse(BPoint(x, y), demoRadius, demoRadius);
+		demoRadius += Settings.lwidth / 2 + Settings.lspace;
+		demoColor.alpha += demoAlphaMod;
+		demoCircle++;
+
+		ConstrainClippingRegion(NULL);
+		SetPenSize(oldPen);
+	}
+	else
+	{
+		//demo background
+		SetHighColor(255, 255, 255, 255);
+		FillRoundRect(rect, 2, 2);
+		//SetHighColor( 0, 0, 0, 255);
+		//StrokeRoundRect( demoRect, 2, 2);
+
+		InitDemo();
+	}
+}
+
+//---------------------------------------------------
+//	Message handler
+//---------------------------------------------------
+void
+DemoView::MessageReceived(BMessage *msg)
+{
+	// drag & drop
+	if (msg->WasDropped())
+	{
+		rgb_color*	color;
+		ssize_t		size;
+		if (msg->FindData("RGBColor", B_RGB_COLOR_TYPE, (const void **) &color, &size) == B_OK)
+		{
+			Settings.color = *color;
+			Invalidate();
+		}
+		return;
+	}
+
+	switch(msg->what)
+	{
+		case WMM_MSG_SET_RADIUS:
+		{
+			msg->FindInt32("be:value", &Settings.radius);
+			Invalidate();
+			break;
+		}
+		case WMM_MSG_SET_CIRCLES:
+		{
+			msg->FindInt32("be:value", &Settings.circles);
+			Invalidate();
+			break;
+		}
+		case WMM_MSG_SET_LWIDTH:
+		{
+			msg->FindInt32("be:value", &Settings.lwidth);
+			Invalidate();
+			break;
+		}
+		case WMM_MSG_SET_LSPACE:
+		{
+			msg->FindInt32("be:value", &Settings.lspace);
+			Invalidate();
+			break;
+		}
+		case WMM_MSG_SET_PULSE:
+		{
+			msg->FindInt32("be:value", &Settings.pulse);
+			Settings.pulse *= 100000;
+			Window()->SetPulseRate(Settings.pulse);
+			Invalidate();
+			break;
+		}
+		default:
+			BView::MessageReceived(msg);
+	}
+}
+
+//---------------------------------------------------
+//	Animate demo circles
+//---------------------------------------------------
+void
+DemoView::Pulse()
+{
+	DrawDemo();
```

Comment:
Thank you for reviewing my task and pointing out the flaws! 

Drawing the demo area in Pulse() is what the original code did, but I will experiment with your suggestion to see if I can make it work without breaking something else. (I was a total novice with Haiku API and had to reply on Skipp_OSX to hand hold me and get me started.)